Support Multiple Current States

blangley28 opened the pull request:
The Neoadjuvant pathway shows an example of multiple options. The pathway now identifies all of them as the current node and expands them.
![Screen Shot 2020-03-24 at 11 31 35 AM](https://user-images.githubusercontent.com/53485517/77445392-68a64980-6dc3-11ea-8b4b-2f9d64550348.png)

Commit:
Update model to use current state list

Commit:
Setup test to include two options for radiation

Commit:
Update Pathway-Results model

Commit:
Update tests to use new PathwayResult model

Commit:
Update OutputResults to handle multiple current states

Commit:
Update index tests

Commit:
Update files to use new PathwayResult data model

Commit:
Fix bug where only one pathway renders

Commit:
Linting

dehall commented:
I just tested with the demo patient, and from the neoadjuvant pathway I selected T > 2 then accepted the TCHP state (the first one on the left). There is the bug with the fhir visualizers but for now we can just comment out all the patient visualizers. What I got looked like this, I would have expected the fact that I accepted one to un-highlight the other 2 in terms of being actionable. Also it's showing an (X) like it was declined instead of being accepted. The arrow is highlighted correctly though
<img width="1354" alt="Screen Shot 2020-03-26 at 7 30 32 AM" src="https://user-images.githubusercontent.com/13512036/77642448-0fa9f300-6f34-11ea-8085-8873e4f1dd59.png">

Commit:
Fix document reference bug

Commit:
Fix pathway cql

Commit:
Fix arrow highlighting to options

Commit:
Remove console.log statements

blangley28 commented:
Fixed issue where remaining options remain marked as current state after selecting an option. Also fixed the bug where accepting the chemotherapy options in the neoadjuvant parkway mark it as declined (pathway + CQL errors). Also fixed bug where the last arrow was not highlighted.

To test remove `<PatientRecord />` on line 130 of `App.tsx`

![Screen Shot 2020-03-27 at 8 31 00 AM](https://user-images.githubusercontent.com/53485517/77756337-59164300-7005-11ea-8b00-d05e9b80f16e.png)
![Screen Shot 2020-03-27 at 8 31 10 AM](https://user-images.githubusercontent.com/53485517/77756343-5ae00680-7005-11ea-88d3-ad2a063b5f66.png)
Note: accepting the chemotherapy regimen in this pathway moves on to the next state because the action here is a MedicationRequest and not a ServiceRequest. The engine handles these differently. Created [PATHWAYS-216](https://jira.mitre.org/browse/PATHWAYS-216) to address it

KeeyanGhoreshi commented on src/components/Graph/Graph.tsx:
```diff
@@ -35,13 +41,13 @@ const Graph: FC<GraphProps> = ({
   const setPath = useCallback(
     (value: PathwayResults): void => {
-      _setPath(value.path);
+      _setPathwayResults(value);
```
Aren't we setting the pathway in two places with this?  Once with this `_setPathwayResults`, and again in `updateEvaluatedPathways`.  The  pathway gets passed to the Graph as a prop already, so updating it in the parent should be enough.

blangley28 replied:
Good catch. Removed the pathwayResults state and updated all uses to use `evaluatedPathway.pathwayResults`

Commit:
Use evaluatedPathway to access pathwayResults

KeeyanGhoreshi commented:
Looks good, will approve with rebase

blangley28 commented:
Rebased

KeeyanGhoreshi commented:
my approval does not count

dehall commented on public/static/pathways/neoadjuvant_pathway.json:
```diff
@@ -161,15 +161,32 @@
         "action": [
+          {
+            "type": "create",
+            "description": "Trastuzumab Medication request",
```
Why does this action say description: "Trastuzumab" but the code below says Doxorubicin?

blangley28 replied:
Probably because I copied and pasted then changed the code but forgot to change the description. Updated.

dehall reviewed (commented):
Overall looks great. Just one question on the pathway

Commit:
Fix Doxorubicin MR description

dehall commented:
Don't forget to update the data model on the wiki

blangley28 commented:
Updated data model on the wiki
